src/engine/runtime.js

```diff
@@ -1218,29 +1222,52 @@ class Runtime extends EventEmitter {
             argJSON.check = argTypeInfo.check;
         }
 
-        const shadowType = (argInfo.menu ?
-            this._makeExtensionMenuId(argInfo.menu, context.categoryInfo.id) :
-            argTypeInfo.shadowType);
-        const fieldType = argInfo.menu || argTypeInfo.fieldType;
+        let valueName;
+        let shadowType;
+        let fieldName;
+        if (argInfo.menu) {
+            const menuInfo = argInfo.menu && context.categoryInfo.menuInfo[argInfo.menu];
+            if (menuInfo.acceptReporters) {
+                valueName = placeholder;
+                shadowType = this._makeExtensionMenuId(argInfo.menu, context.categoryInfo.id);
+                fieldName = argInfo.menu;
+            } else {
+                argJSON.type = 'field_dropdown';
+                argJSON.options = menuInfo.items;
+                valueName = null;
+                shadowType = null;
+                fieldName = placeholder;
+            }
+        } else {
+            valueName = placeholder;
+            shadowType = argTypeInfo.shadowType;
+            fieldName = argTypeInfo.fieldType;
+        }
 
         // <value> is the ScratchBlocks name for a block input.
-        context.inputList.push(`<value name="${placeholder}">`);
+        if (valueName) {
+            context.inputList.push(`<value name="${placeholder}">`);
+        }
 
         // The <shadow> is a placeholder for a reporter and is visible when there's no reporter in this input.
         // Boolean inputs don't need to specify a shadow in the XML.
         if (shadowType) {
             context.inputList.push(`<shadow type="${shadowType}">`);
+        }
 
-            // <field> is a text field that the user can type into. Some shadows, like the color picker, don't allow
-            // text input and therefore don't need a field element.
-            if (fieldType) {
-                context.inputList.push(`<field name="${fieldType}">${defaultValue}</field>`);
-            }
+        // <field> is a text field that the user can type into. Some shadows, like the color picker, don't allow
```

Comment:
This comment is actually not strictly correct... especially now that we have non-droppable menus...